### ISSUE TITLE
Export function for per-image calcification rates

### ISSUE THREAD

**Issue by StephenChan** [beijbom/coralnet#369]
Prerequisite: #368

Be able to export a CSV of per-image calcification rates for a source.

Example calculation:

- Suppose an image has 10 points. 5 CCA, 3 Porites, 2 Acropora.
- Suppose those three labels' calcification rates are taken to be 0.4, 7.2, and 18.0 respectively.
- The image's calcification rate is then: `(5*0.4 + 3*7.2 + 2*18.0) / 10 = 5.96`

This export function can live alongside the other data exports, which currently reside in the Browse Images page, Image Actions box. It should be possible to use the Browse Images search box to filter to a subset of images, and then export calcification rate data on that image subset. (I'll make another issue about moving all the data export functions to their own easier-to-find page, while retaining the image filtering capability.)

Initially, there can be two radio-button options for selecting label calcification rates to use in the calculation:

- Use the Indo-Pacific default rates
- Use the Atlantic default rates

The default rates are defined by #368. Later there can be a third option for providing custom rates, but I'll make a separate issue for that.

The text for each of the above two options can include a download link to a CSV of the default rates (for that source's labelset), for the user's information. This will also help for the custom-rates step later.

If a particular label in the labelset doesn't have a rate specified, default to a rate of 0 for that label.

Main contents of the exported CSV - one row per image, including the following info:

- Image ID (for example, 12904).

- Image name (for example, IMG_3650.JPG).

- Overall calcification rate for the image. Three columns: Mean, lower bound, and upper bound. Values can have many decimal places.

- (Optional, on by default?) One column per label in the source's labelset, saying the mean calcification rate based on that label. This can make the CSV quite large depending on the source's labelset size; some sources have over 100 labels. Values can generally have as many decimal places as the label rate-tables do.

- (Optional, off by default) Two additional columns per label in the labelset, saying the lower and upper confidence bounds of the rate based on that label. So if a source has 100 labels, there would be around 300 columns total.

Also see #373 (summary stats) and #374 (meta info). Could work on this issue either before or after those issues, or partly at the same time.

The exported CSV filename can include the name of the source and the date of the export action. For example, "My Source - Calcification rates - 2021-03-22.csv"

There should be checkboxes in the export form that let people specify which optional information they want.

**Comment by traviscourtney** [comment]
We previously discussed using the user inputted latitude and longitude to determine whether the Atlantic or Indo-Pacific rate table should be used. I have bounding boxes for the respective ocean basins that could be used as a lookup table for whether the coordinates are in the Atlantic vs Indo-Pacific. I think the radio buttons are a solid option for now, but future classification based on the coordinates would be a really nice feature if there's a seemingly straightforward implementation within the CoralNet framework?

It would be really great for any exported calcification rates to include a column with the name for the rate table used. For example having the image calcification rates tied to the the default Atlantic version 1.0 rates, Indo-Pacific version 1.0 rates, or custom user inputted table (with version ID name) in this way would improve traceability with the rates used to determine the calcification for the respective image.

**Comment by StephenChan** [comment]
For automatically determining which rate table should be used, I don't think there's a straightforward lat/long based way that can be implemented directly in CoralNet anytime soon. The next best idea I could think of was #389. Basically a source owner would have to use a script outside of CoralNet to read the lat/long of each image, and accordingly craft an auxiliary-metadata CSV file which could be uploaded to the CoralNet source. From there, the idea in #389 could be used to apply one table or another based on images' aux. metadata values.

And you're right, if multiple rate tables can be used in a single export, then there should be a column specifying the rate table that's used for each image.

**Comment by StephenChan** [comment]
Done in PR #404.
